lib/src/Pos/Diffusion/Full/Block.hs

```diff
@@ -704,9 +704,12 @@ handleStreamStart logTrace logic oq = listenerConv logTrace oq $ \__ourVerInfo n
                       lift $ traceWith logTrace (Debug, sformat ("handleStreamStart:loop new window "%shown%" from "%build) u nodeId)
                       loop nodeId conv (msuWindow u)
     loop nodeId conv window = do
-        b <- await
-        lift $ sendRaw conv $ serializeMsgStreamBlock $ MsgSerializedBlock b
-        loop nodeId conv (window - 1)
+        mB <- await
+        case mB of
```

Comment:
❗️ The `await` from `conduit` returns a `(Maybe a)`. The `await` from `pipes` returns an `a`.

The conduit docs say [here](https://www.stackage.org/haddock/lts-11.7/conduit-1.3.0.2/Data-Conduit.html#v:await):

> Wait for a single input value from upstream. If no data is available, returns Nothing. Once await returns Nothing, subsequent calls will also return Nothing.

It seems like if/when we hit the `Nothing` case, we will loop forever. However in the previous `pipes` code, I believe the `Nothing` values would just be dropped, so *I think* the behavior should be the same.

Reply:
With conduit, `await` returning a `Nothing` means there is no more data to be got from the source.

I there for think the we should *not* loop there. Instead, the `Nothing` case should just do `pure ()` or `pure someReturnValue` depending on what is required.

Reply:
This chunk is in `IO ()`, so `pure ()` seems appropriate.